src/factory/clients.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  BitcoinEsploraBatchBaseProvider,
+  BitcoinFeeApiProvider,
+  BitcoinHDWalletProvider,
+  BitcoinSwapEsploraProvider,
+} from '@liquality/bitcoin';
+import { Client, Fee } from '@liquality/client';
+import { ChainId } from '@liquality/cryptoassets';
+import { EvmChainProvider, EvmSwapProvider, EvmWalletProvider } from '@liquality/evm';
+import { NearChainProvider, NearSwapProvider, NearTypes, NearWalletProvider } from '@liquality/near';
+import { SolanaChainProvider, SolanaWalletProvider } from '@liquality/solana';
+import { TerraChainProvider, TerraSwapProvider, TerraTypes, TerraWalletProvider } from '@liquality/terra';
+import { Network as ChainifyNetwork } from '@liquality/types';
+import buildConfig from '../build.config';
+import { AccountType, Asset, Network } from '../store/types';
+import cryptoassets from '../utils/cryptoassets';
+import { LEDGER_BITCOIN_OPTIONS } from '../utils/ledger';
+import { ChainNetworks } from '../utils/networks';
+import { walletOptionsStore } from '../walletOptions';
+
+export function createBtcClient(
+  network: Network,
+  mnemonic: string,
+  accountType: AccountType,
+  baseDerivationPath: string
+) {
+  const isTestnet = network === 'testnet';
+  const bitcoinNetwork = ChainNetworks.bitcoin[network];
+  const esploraApi = buildConfig.exploraApis[network];
+  const batchEsploraApi = buildConfig.batchEsploraApis[network];
+
+  const feeProvider = new BitcoinFeeApiProvider('https://liquality.io/swap/mempool/v1/fees/recommended');
+  const chainProvider = new BitcoinEsploraBatchBaseProvider(
+    {
+      batchUrl: batchEsploraApi,
+      url: esploraApi,
+      network: bitcoinNetwork,
+      numberOfBlockConfirmation: 2,
+    },
+    feeProvider
+  );
+  const swapProvider = new BitcoinSwapEsploraProvider({
+    network: bitcoinNetwork,
+    scraperUrl: esploraApi,
+  });
+
+  // TODO: make sure Ledger works
+  if (accountType.includes('bitcoin_ledger')) {
+    const option = LEDGER_BITCOIN_OPTIONS.find((o) => o.name === accountType);
+    if (!option) {
+      throw new Error(`Account type ${accountType} not an option`);
+    }
+    const { addressType } = option;
+    if (!walletOptionsStore.walletOptions.createBitcoinLedgerProvider) {
+      throw new Error('Wallet Options: createBitcoinLedgerProvider is not defined - unable to build ledger client');
+    }
+    const ledgerProvider = walletOptionsStore.walletOptions.createBitcoinLedgerProvider(
+      network,
+      bitcoinNetwork,
+      addressType,
+      baseDerivationPath
+    );
+    swapProvider.setWallet(ledgerProvider);
+  } else {
+    const walletOptions = {
+      network: bitcoinNetwork,
+      baseDerivationPath,
+      mnemonic,
+    };
+    const walletProvider = new BitcoinHDWalletProvider(walletOptions, chainProvider);
+    swapProvider.setWallet(walletProvider);
+  }
+
+  if (isTestnet) {
+    chainProvider.setFeeProvider(null);
+  }
+
+  return new Client().connect(swapProvider);
```

Comment:
Kinda weird how only swap provider can use connect here
in the ideal way, none of the providers need to take other providers as argument, or need to set it eg.`setFeeProvider`

at the end you just do `return new Client().connect(swapProvider).connect(feeProvider).connect(chainProvider).connect(walletProvider) and that's it

Reply:
`new Client().connect(swapProvider);` is the same as new `Client().connect(swapProvider).connect(chainProvider).connect(walletProvider)`